### PR TITLE
Add unit test for move custom remote root

### PR DESCRIPTION
**Pull request by FlexW**
Signed-off-by: Felix Weilbach <felix.weilbach@nextcloud.com>

### CODE REVIEW

**Comment by allexzander** [comment]
@FlexW It could use some refactoring but I like this approach and the fact that it's short and does the job. Nice.

**Review by mgallien — APPROVED** [review]

File: test/syncenginetestutils.cpp [hunk 1]
```diff
@@ -288,7 +288,11 @@ FakePropfindReply::FakePropfindReply(FileInfo &remoteRootFileInfo, QNetworkAcces
     auto writeFileResponse = [&](const FileInfo &fileInfo) {
         xml.writeStartElement(davUri, QStringLiteral("response"));
 
-        xml.writeTextElement(davUri, QStringLiteral("href"), prefix + QString::fromUtf8(QUrl::toPercentEncoding(fileInfo.path(), "/")));
+        QString url = prefix + QString::fromUtf8(QUrl::toPercentEncoding(fileInfo.path(), "/"));
+        if (!url.endsWith(QChar('/'))) {
```
Comment (mgallien) [inline test/syncenginetestutils.cpp:292]: why is this change really needed ?

Comment (FlexW) [reply test/syncenginetestutils.cpp:292]: otherwise, an assert will be triggered during the discovery phase because the URL is not in the expected format. probably it was before no problem because we didn't had subfolders.

**Review by allexzander — APPROVED** [review]
